### PR TITLE
Fix #36 - Async-ify postinstall

### LINKED ISSUE #36: Performance: 'asyncify' postinstall tasks
When we unpack the package, our `postinstall` is pretty slow - we're _synchronously_ running all of our tasks (restoring hardlinks and symlinks). We should switch these over to use the async APIs and run them in parallel - would be a major speed up on install!

### PR DESCRIPTION
**Pull request by bryphe**

### CODE REVIEW

**Comment by ulrikstrid** [comment]
Are we okay with this only being runnable on node 8+?

**Comment by andreypopp** [comment]
@ulrikstrid 

> Are we okay with this only being runnable on node 8+?

I think it's reasonable 10 is LTS now.

File: scripts/consolidate-links.js [hunk 1]
```diff
@@ -24,20 +32,20 @@ const consolidateLinks = () => {
         fs.mkdirSync(path.join(cygwinFolder, "_links"));
     }
 
-    const deleteFile = (file) => {
+    const deleteFile = async (file) => {
         const fileToDelete = path.join(cygwinFolder, path.normalize(file.trim()));
         console.log(`Deleting: ${fileToDelete}`);
         if (!fs.existsSync(fileToDelete)) {
```
Comment (andreypopp) [inline scripts/consolidate-links.js]: `existsSync` here: can be replace with async version

File: scripts/consolidate-links.js [hunk 2]
```diff
@@ -77,27 +88,29 @@ const restoreLinks = () => {
...
 
         const src = path.join(cygwinFolder, "_links", getNameForKey(key));
 
-        l.forEach((file) => {
+       let promises = l.map(async (file) => {
             const dst = path.join(cygwinFolder, file.trim());
-            ensureFolder(path.dirname(dst));
+            await ensureFolder(path.dirname(dst));
 
             if (fs.existsSync(dst)) {
```
Comment (andreypopp) [inline scripts/consolidate-links.js]: `existsSync` here: can be replace with async version

File: scripts/consolidate-links.js [hunk 3]
```diff
@@ -117,15 +130,16 @@ const restoreLinks = () => {
 
         if (fs.existsSync(link)) {
```
Comment (andreypopp) [inline scripts/consolidate-links.js]: existsSync here: can be replace with async version

File: scripts/consolidate-links.js (hunk 1 above)
Comment (bryphe) [reply scripts/consolidate-links.js]: Thanks @andreypopp ! Good point.

**Comment by bryphe** [comment]
Thanks @andreypopp and @ulrikstrid for reviewing! Much faster now after fixing the remaining synchronous calls and fixing an issue with `promisify(cp.spawn)`. I'll bring in if CI passes.

File: postinstall.js [hunk 4]
```diff
@@ -16,3 +16,8 @@ if (!fs.existsSync(path.join(__dirname, ".cygwin"))) {
 console.log("Restoring hardlinks...");
 restoreLinks();
 console.log("Complete!");
```
Comment (andreypopp) [inline postinstall.js]: This message will be printed before restoreLinks completes.

Comment (bryphe) [reply postinstall.js]: Good catch!

**Comment by bryphe** [comment]
Phew, CI finally passing! 💯 Will bring this in now.

**Pull request merged**